src/Slices/Notification/UI/Drawer/Item.tsx

```diff
@@ -23,18 +22,31 @@ interface Props {
 }
 
 export const Item: React.FC<Props> = ({ notification, onUpdate }) => {
+  const { routeManager } = useContext(DependencyContext);
+  const detailsLink: RouteKindWithId<"CompileDetails"> | undefined =
+    routeManager.getUrlCompileDetailsId(
+      notification.uri
+    ) as RouteKindWithId<"CompileDetails">;
```

Comment:
Since you have a check later to see if it's defined, there's no need for the type cast I guess?